src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -589,8 +589,13 @@ private function registerProfilerConfiguration(array $config, ContainerBuilder $
             }
         }
 
-        if (!$config['collect']) {
-            $container->getDefinition('profiler')->addMethodCall('disable', array());
+        $profilerDefinition = $container->getDefinition('profiler');
+        $profilerDefinition->addTag('kernel.reset', array('method' => 'reset'));
+        if ($config['collect']) {
+            $profilerDefinition->addTag('kernel.reset', array('method' => 'enable'));
```

Comment:
Can't this be handled in the reset method itself instead? We should try harder to *not* have to call several methods to reset (and thus revert the multi-tags possibility).

Reply:
If I wanted to do this, the profiler would need to know its initial state (enabled or disabled) so it can reset itself to that state. I don't feel comfortable with saving that information inside the profiler.

Reply:
ok, still look suspicious to me:
in a kernel loop, re-enabling at reset time means collecting timings from end of previous request to end of next request, isn't it? But next request might come very long after, so timings won't mean anything.
Btw, the Twig profiler has the same issue, don't you think? I mean, calling "enter()" at reset time also?
Shouldn't this re-enabling happen at kernel.request time instead?